### ISSUE TITLE
Expose full extra-files support, or at least GenericJson updater

### ISSUE THREAD

**Issue by scop** [google-github-actions/release-please-action#661]
### TL;DR

The currently available generic updater for extra-files applies to many cases using the marker regexps.

However, applying it to JSON is quite awkward, and even more so when the JSON in question does not allow arbitrary extra properties beyond the expected schema. In some cases, such as the one I have at hand, it can be practically impossible.

Exposing the GenericJson updater would be good to have.

### Detailed design

Backwards compatibility with the current extra-files content syntax might be problematic. But I suppose some kind of content sniffing would work, although it wouldn't be pretty -- the extra-files content could be parsed as JSON if it begins with `^\s*[`, and the parsed payload passed onwards to support the full release-please extra files payload, including the generic JSON (and XML) updaters. There is a precedent for passing JSON content as string in `changelog-types`.

**Comment by scop** [comment]
Indeed interested in that, and thanks for the info, will try it out sometime.

However, if the manifest driven way is highly recommended, I suggest addressing the following issues in README.md:
- There's no mention that manifest driven is recommended, it's just documented as an alternative with easier bootstrapping.
- There's no mention that manifest driven provides access to most (all) release-please features, the other way does not.
- `command` is documented as "... either github-release, or release-pr, manifest, manifest-pr (defaults to running both)"; but what does "both" here mean -- which both of the four listed?

Also, a complete example of something using release-please-action with the manifest stuff would be most welcome.

**Comment by henry-brisson** [comment]
Unfortunately, the manifest approach doesn't seem to work for me. 

My case: I have a _lot_ of Node repos that use a single shared workflow for ReleasePlease:

```yaml
name: Release (Shared)
on:
  workflow_call:
jobs:
  manage-release:
    runs-on: ubuntu-latest
    steps:
      - name: Create/Update Release PR
        uses: google-github-actions/release-please-action@v3.7.3
        id: release-pr
        with:
          release-type: node
          include-v-in-tag: false
          extra-files: |
            README.md
            version.js
            src/version.js
            config/version.js
```

And then in my projects I just use it like this:

```yaml
name: Release
on:
  push:
    branches: main
jobs:
  default:
    uses: <public-github-url>/.github/workflows/default-release.yml
```

I came with the setup above so that all projects that use ReleasePlease use the same settings, which include `extra-files`. This been working really great so far, but in the current state, I'm unable to easily parse JSON files with the limited `extra-files` capability from this action.

👉 Creating manifest files for each repo would be highly counter-productive, as I'm aiming to simplify and reuse configuration across multiple repos, so the less files the better.

It would make more sense if `extra-files` wasn't a multiline string, but a real YAML array like:

```yaml
extra-files:
- README.md
- version.js
- src/version.js
- config/version.js
- ...
```

which would then allow an object to be part of the array:

```yaml
extra-files:
- README.md
- version.js
- src/version.js
- config/version.js
- type: json
  path: path/to/file.json
  jsonpath: "$.json.path.to.field"
- type: xml
  path: "path/to/file.xml"
  xpath: "//xpath/to/field"
```

Then the parser used by manifest files could be reused by this action as well. Hope it all makes sense!
